src/components/TodoApp/TodoList.jsx

```diff
@@ -0,0 +1,33 @@
+import { useMemo } from 'react';
+
+import TodoItem from './TodoItem';
+
+/**
+ * @param {{ todos: Todo[], onComplete: (id: string) => void }} props
+ */
+const TodoList = ({ todos = [], onComplete }) => {
+  const TodoListTitle = useMemo(() => () => <h2 style={{ display: 'none' }}>Todo 리스트</h2>, []);
+
+  if (todos.length === 0) {
+    return (
+      <div>
+        <TodoListTitle />
+        <p>할 일이 없어요!</p>
+      </div>
+    );
+  }
+  return (
+    <div>
+      <TodoListTitle />
+      <ul style={{ listStyle: 'decimal' }}>
+        {todos.map((todo) => (
+          <li key={todo.id}>
+            <TodoItem todo={todo} onComplete={onComplete} />
+          </li>
```

Comment:
```suggestion
const Title = () => {
  return (
    <h2 style={{ display: 'none' }}>Todo 리스트</h2>
  );
};

const TodoList = ({ todos = [], onComplete }) => {
  if (todos.length === 0) {
    return (
      <div>
        <TodoListTitle />
        <p>할 일이 없어요!</p>
      </div>
    );
  }
  
  return (
    <div>
      <TodoListTitle />
      <ul style={{ listStyle: 'decimal' }}>
        {todos.map((todo) => (
          <li key={todo.id}>
            <TodoItem todo={todo} onComplete={onComplete} />
          </li>
```

이렇게 밖으로 빼줘도 될 것 같아요!

Reply:
한 파일에 컴포넌트 여러개를 선언하는게 약간 걸려서 TodoList 컴포넌트 안에 넣어줬는데 밖으로 추출하겠습니다:)